galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -49,15 +50,17 @@ const defaultSnapshotGroup = "default"
 const syntheticSnapshotGroup = "syntheticServiceEntry"
 
 // NewAnalyzingDistributor returns a new instance of AnalyzingDistributor.
-func NewAnalyzingDistributor(u StatusUpdater, a analysis.Analyzer, d Distributor, cr CollectionReporterFn) *AnalyzingDistributor {
+func NewAnalyzingDistributor(u StatusUpdater, a *analysis.CombinedAnalyzer, d Distributor, cr CollectionReporterFn,
+	disabledCollections collection.Names, xformProviders transformer.Providers) *AnalyzingDistributor {
```

Comment:
Can we make the calculation for disabled resources on the outside? It shouldn't be the responsibility of the analyzing distrbutor, especially given that the CombinedAnalyzer is being explicitly passed.

Also, what happens when Galley restarts and the set of available collections change (due to a CRD update)?

Reply:
> Can we make the calculation for disabled resources on the outside?

Yes, it made more sense here before I made CombinedAnalyzer explicitly passed. Will fix.

> Also, what happens when Galley restarts and the set of available collections change (due to a CRD update)?

As I'm reading things it should be fine: Processing2.Start will create a new AnalyzingDistributor instance (as well as new inputs going into it).

Reply:
What happens when some CRDs are not available when Galley starts, but appear 30 minutes later (and Galley restarts its processing pipeline?).

Reply:
As I understand it (please correct where I'm wrong):

* Galley starts
* Processing2.Start() fires, discovers kubeResources, creates an AnalyzingDistributor
* Galley starts doing analysis & status updates
* CRD update happens
* Processing2.Stop() fires, kills everything
* Processing2.Start() fires, rediscovers kubeResources (including the update now), creates a new AnalyzingDistributor (with the new kubeResources)
* Galley starts doing analysis & status updates (including the new CRDs that it previously didn't have).

Reply:
As discussed offline: Actually Processing doesn't start/stop if there is a meshconfig update, but in the current implementation that doesn't matter since analyzer disabling is entirely dependent on the metadata (which wouldn't be changing). 

If we need the set of enabled analyzers to be more dynamic, then we would need to be more clever and probably use some sort of event-based model (possibly using a special collection object that holds metadata like which analyzers to disable, and passing that through). But we don't have a use case for that yet.